backend/src/auth/auth.controller.ts

```diff
@@ -0,0 +1,77 @@
+import {
+	BadRequestException,
+	Body,
+	Controller,
+	Post,
+	Res,
+	UseGuards,
+	UseInterceptors,
+	ClassSerializerInterceptor,
+	Get,
+	Req,
+	HttpCode,
+	HttpStatus,
+} from '@nestjs/common';
+import { UserService } from '../user/user.service';
+import * as bcrypt from 'bcrypt';
+import { JwtService } from '@nestjs/jwt';
+import { Response, Request } from 'express';
+import { AuthGuard } from './auth.guard';
+import { AuthService } from './auth.service';
+import { LoginUserDto } from './dto/login-user.dto';
+
+@UseInterceptors(ClassSerializerInterceptor)
+@Controller()
+export class AuthController {
+	constructor(
+		private userService: UserService,
+		private jwtService: JwtService,
+		private authService: AuthService,
+	) {}
+
+	@Post('login')
+	async login(
+		@Body() loginUserDto: LoginUserDto,
+		@Res({ passthrough: true }) response: Response,
+	) {
+		try {
+			const user = await this.userService.findOne({
+				where: {
+					name: loginUserDto.name,
+				},
+			});
+			if (
+				!user ||
+				!(await bcrypt.compare(loginUserDto.password, user.password))
+			) {
+				throw new BadRequestException('Invalid user/password combination');
+			}
+			const jwt = await this.jwtService.signAsync({ id: user.id });
+			response.cookie('jwt', jwt, { httpOnly: true });
+			return user;
+		} catch (e) {
+			throw new BadRequestException('Invalid user/password combination');
+		}
+	}
+
+	@UseGuards(AuthGuard)
+	@Post('logout')
+	@HttpCode(HttpStatus.OK)
+	async logout(@Res({ passthrough: true }) response: Response) {
+		response.clearCookie('jwt');
+		return {
+			message: 'Success',
```

Comment:
Good one! Technically, I guess, yes, it could fail...
source of clearCookie form expressjs: https://github.com/expressjs/express/blob/158a17031a2668269aedb31ea07b58d6b700272b/lib/response.js#L824:L887

It has a `throw`, but only for signed cookies.. 
And it uses JSON.stringify, which could also throw.. but unlikely in this use case [(throws when you give it a object with circular reference or when out of memory..)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)

The case against protecting against this is that expressjs in their own tests, do not check for errors:
https://github.com/expressjs/express/blob/a65913776d0b16837364ee66caa1a7f38a9997c0/test/res.clearCookie.js

But that could be a lack of testing from their side..... who knows 🤷